rir/src/compiler/analysis/abstract_value.h

```diff
@@ -175,15 +176,15 @@ struct AbstractREnvironment {
         }
 
         std::unordered_set<Value*> fps;
-        for (auto e : functionPointers)
+        for (auto e : mkClosures)
             fps.insert(std::get<0>(e));
-        for (auto e : other.functionPointers)
+        for (auto e : other.mkClosures)
             fps.insert(std::get<0>(e));
         for (auto n : fps) {
-            if (functionPointers[n] != UnknownFunction &&
-                (other.functionPointers.count(n) == 0 ||
-                 functionPointers[n] != other.functionPointers.at(n))) {
-                functionPointers[n] = UnknownFunction;
+            if (mkClosures[n] != UnknownFunction &&
+                (other.mkClosures.count(n) == 0 ||
```

Comment:
What does this mean? Here we merge two aenvs, and one of them has a function that is known, and the other doesn't have it. Why is the result unknown and not the one we know?

Reply:
UnknownFunction is the top. The lattice is

    Top: UnknownFunction
                 |
      MkClosureFun*
                 |
     Bottom:  Absent in the hashmap

Reply:
oh, I see what you mean, yeah from that lattice it should be possible to set to the other